debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/commands/ToggleInstructionStepModeHandler.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Advantest Europe GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * 				Raghunandana Murthappa
+ *******************************************************************************/
+package org.eclipse.cdt.debug.internal.ui.commands;
+
+import org.eclipse.cdt.debug.core.CDebugCorePlugin;
+import org.eclipse.cdt.debug.core.model.ISteppingModeTarget;
+import org.eclipse.cdt.debug.internal.ui.actions.CDTDebugPropertyTester;
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.Command;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.commands.State;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.IPreferenceChangeListener;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.PreferenceChangeEvent;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.ui.handlers.RegistryToggleState;
+
+/**
+ * Handles the command org.eclipse.cdt.debug.internal.ui.actions.ToggleInstructionStepModeCommand
+ * Turns instruction step mode on/off for selected target.
+ * @author Raghunandana Murthappa
+ */
+public class ToggleInstructionStepModeHandler extends AbstractHandler implements IPreferenceChangeListener {
+
+	public static final String INSTRUCTION_STEPPING_MODE = "instruction_stepping_mode"; //$NON-NLS-1$
```

Comment:
`ITargetProperties` is not used anymore. `DsfSteppingModeTarget` was the only client and now it does not implement it. I will remove it.

Reply:
How do you know that `DsfSteppingModeTarget` is the only user or a public interface?
It might well be the only one implementing the interface in the CDT code base, but it might be implemented by downstream too, and with the removal of the `ITargetProperties` interface, their builds will now be broken.

For the CDT 12 release, I assume that it could be fine to remove it if we really want to as it's a major release anyway.